pkg/agent/stream/watchtopic.go

```diff
@@ -142,54 +240,69 @@ type WatchTopicValues struct {
 			{GroupKind: mv1.CredentialGVK().GroupKind, ScopeName: scope, EventTypes: createdOrUpdated},
 			{GroupKind: mv1.AccessRequestGVK().GroupKind, ScopeName: scope, EventTypes: createdOrUpdated},
 			{GroupKind: mv1.ManagedApplicationGVK().GroupKind, ScopeName: scope, EventTypes: createdOrUpdated},
 			{GroupKind: mv1.CredentialRequestDefinitionGVK().GroupKind, ScopeName: scope, EventTypes: all},
 			{GroupKind: mv1.AccessRequestDefinitionGVK().GroupKind, ScopeName: scope, EventTypes: all},
-		},
+		}...)
+	}
+	return WatchTopicValues{
+		Name:        name,
+		Title:       name,
+		Description: fmt.Sprintf(desc, "discovery", scope),
+		Kinds:       kinds,
 	}
 }
 
 // NewTraceWatchTopic creates a WatchTopic template string
-func NewTraceWatchTopic(name, scope string, agentResourceGroupKind v1.GroupKind) WatchTopicValues {
+func NewTraceWatchTopic(name, scope string, agentResourceGroupKind v1.GroupKind, features watchTopicFeatures) WatchTopicValues {
+	kinds := []kindValues{
+		{GroupKind: agentResourceGroupKind, ScopeName: scope, EventTypes: updated},
+		{GroupKind: mv1.APIServiceGVK().GroupKind, ScopeName: scope, EventTypes: all},
+		{GroupKind: mv1.APIServiceInstanceGVK().GroupKind, ScopeName: scope, EventTypes: all},
+	}
+	if features.IsMarketplaceSubsEnabled() {
+		kinds = append(kinds, []kindValues{
+			{GroupKind: mv1.AccessRequestGVK().GroupKind, ScopeName: scope, EventTypes: all},
+			{GroupKind: mv1.ManagedApplicationGVK().GroupKind, ScopeName: scope, EventTypes: all},
+		}...)
+	}
 	return WatchTopicValues{
 		Name:        name,
 		Title:       name,
 		Description: fmt.Sprintf(desc, "traceability", scope),
-		Kinds: []kindValues{
-			{GroupKind: agentResourceGroupKind, ScopeName: scope, EventTypes: updated},
-			{GroupKind: mv1.APIServiceGVK().GroupKind, ScopeName: scope, EventTypes: all},
-			{GroupKind: mv1.APIServiceInstanceGVK().GroupKind, ScopeName: scope, EventTypes: all},
-			{GroupKind: mv1.AccessRequestGVK().GroupKind, ScopeName: scope, EventTypes: all},
-			{GroupKind: mv1.ManagedApplicationGVK().GroupKind, ScopeName: scope, EventTypes: all},
-		},
+		Kinds:       kinds,
 	}
 }
 
 // NewGovernanceAgentWatchTopic creates a WatchTopic template string
-func NewGovernanceAgentWatchTopic(name, scope string, agentResourceGroupKind v1.GroupKind) WatchTopicValues {
+func NewGovernanceAgentWatchTopic(name, scope string, agentResourceGroupKind v1.GroupKind, features watchTopicFeatures) WatchTopicValues {
+	kinds := []kindValues{
+		{GroupKind: agentResourceGroupKind, ScopeName: scope, EventTypes: updated},
+		{GroupKind: mv1.AmplifyRuntimeConfigGVK().GroupKind, ScopeName: scope, EventTypes: all},
+		{GroupKind: mv1.APIServiceGVK().GroupKind, ScopeName: scope, EventTypes: all},
+		{GroupKind: mv1.APIServiceInstanceGVK().GroupKind, ScopeName: scope, EventTypes: all},
+	}
+	if features.IsMarketplaceSubsEnabled() {
+		kinds = append(kinds, []kindValues{
+			{GroupKind: mv1.AccessRequestGVK().GroupKind, ScopeName: scope, EventTypes: all},
```

Comment:
AccessRequest filter for Governance agent is needed irrespective of agent feature. The agent will register a proxy handler to receive events for AccessRequest. 
When marketplace provisioning feature is disabled, since the SDK AccessRequest handler is not registered that should ensure the to not process AccessRequest for provisioning.